packages/core/src/getNextSnapshot.ts

```diff
@@ -0,0 +1,74 @@
+import { createEmptyActor } from './actors/index.ts';
+import {
+  ActorRefFrom,
+  ActorScope,
+  AnyActorLogic,
+  AnyActorScope,
+  EventFromLogic,
+  InputFrom,
+  SnapshotFrom
+} from './types.ts';
+
+/** @internal */
+export function createInertActorScope<T extends AnyActorLogic>(
+  _actorLogic: T
+): AnyActorScope {
+  const self = createEmptyActor() as ActorRefFrom<T>;
+  const inertActorScope: ActorScope<SnapshotFrom<T>, EventFromLogic<T>, any> = {
+    self,
+    defer: () => {},
+    id: '',
+    logger: () => {},
+    sessionId: '',
+    stopChild: () => {},
+    system: (self as any).system // TODO: fix types?
```

Comment:
This is caused by `T extends AnyActorLogic` pattern. Or rather with the `ActorRefFrom<T>` pattern. It's fundamentally at odds with how TS works and for that reason, I'm, at the moment, not a fan of this pattern.

The problem is that TS can't guarantee that `ActorRefFrom<T>` "returns" an `ActorRef` type. It's a conditional type and it got fed up generic type argument - as such it stays deferred and can't be "resolved". And well, you can't access properties on a conditional type :v

Reply:
What would be the solution?

Reply:
Stop using the `ActorRefFrom<T>` pattern ;p While separate generics are somewhat annoying - they fit the type system better. If we don't want to do this here then we have to be fine with this `as any` and we should remove this TODO comment - since it's not something that can be fixed while using the `ActorRefFrom<T>` pattern

Reply:
Removed